src/main/java/seedu/address/model/VersionedConcierge.java

```diff
@@ -8,7 +8,7 @@
  */
 public class VersionedConcierge extends Concierge {
 
-    private final List<ReadOnlyConcierge> conciergeStateList;
```

Comment:
Do you need to remove `final` here?

Reply:
Hey sorry, I removed it previously because I didn't know the `removeStatesAfterCurrentPointer` method existed. Will fix this.